extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ResourceNotFoundHandler.java

```diff
@@ -23,7 +23,13 @@ public void handle(RoutingContext routingContext) {
         if (header != null && header.startsWith("application/json")) {
             handleJson(routingContext);
         } else {
-            handleHTML(routingContext);
+            // If not explicitly asked for json, let determine based on the user agent
+            String userAgent = routingContext.request().getHeader("User-Agent");
+            if (userAgent != null && (userAgent.contains("Wget") || userAgent.contains("curl"))) {
```

Comment:
```suggestion
            if (userAgent != null && (userAgent.toLowercase(Locale.ROOT).startsWith("wget/") || userAgent.toLowercase(Locale.ROOT).startsWith("curl/"))) {
```

Reply:
Thanks @gsmet - I applied your suggestion. Also note that the Accept header is always used if it's available, but if not it will look at the user agent, and decide between text and html